### PR TITLE
chore(storage): align storage category exception types with other platforms

### PR DESCRIPTION
**Pull request by HuiSF**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_core/lib/src/types/exception/storage/key_not_found_exception.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_core.storage.key_not_found_exception}
+/// Exception thrown when the key is not found in the storage.
```
Comment (ragingsquirrel3) [inline packages/amplify_core/lib/src/types/exception/storage/key_not_found_exception.dart:7]: question: In the case of s3 does this mean that the key is not in the bucket? I found word "storage" a little confusing here bc wasn't sure if it referred to local/cloud storage. Are these comments/descriptions part of cross-platform error definitions or can we change them?

If we can change it I dk if can be changed to clarify like "... not found in the storage service."

**Review by ragingsquirrel3 — APPROVED** [review]
approved with 1 question/nit

File: packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart [hunk 2]
```diff
@@ -196,4 +142,84 @@ class S3Exception extends StorageException {
...
   }
+
+  /// A [StorageLocalFileNotFoundException] thrown when download destination file
+  /// path points to a directory.
+  static const StorageLocalFileNotFoundException
+      downloadDestinationFilePathIsDirectory =
+      StorageLocalFileNotFoundException(
+    'Download destination file path is a directory',
+    recoverySuggestion:
+        'Ensure the path provided with `AWSFile` is pointing to a file, but not a directory.',
```
Comment (Equartey) [inline packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart:153]:
```suggestion
        'Ensure the path provided with `AWSFile` is pointing to a file, and not a directory.',
```
nit: a word

File: packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart [hunk 3]
```diff
@@ -196,4 +142,84 @@ class S3Exception extends StorageException {
...
+  );
+
+  /// A [StorageLocalFileNotFoundException] thrown when download destination file
+  /// path is `null`.
+  static const StorageLocalFileNotFoundException
+      downloadDestinationFilePathIsNull = StorageLocalFileNotFoundException(
+    'Download destination file path is null',
+    recoverySuggestion: 'Ensure to provide a valid file path with `AWSFile`.',
```
Comment (Equartey) [inline packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart:161]:
```suggestion
    recoverySuggestion: 'Ensure a valid file path with `AWSFile` was provided.',
```
nit: wording

Comment (Equartey) [inline packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart:161]:
```suggestion
    recoverySuggestion: 'Ensure a valid file path with `AWSFile` is provided.',
```
nit: wording

**Review by Equartey — APPROVED** [review]
Looks good, just a couple grammar nits.

File: packages/amplify_core/lib/src/types/exception/storage/key_not_found_exception.dart (hunk 1 above)
Comment (HuiSF) [reply packages/amplify_core/lib/src/types/exception/storage/key_not_found_exception.dart:7]: Yea the specific message and recovery suggestion can be customized when initiate this exception according to specific occasion. storage service sounds good to me. 👍

**Pull request merged**